adapters/risemediatech/risemediatech_test.go

```diff
@@ -0,0 +1,162 @@
+package risemediatech
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/prebid/prebid-server/v3/adapters/adapterstest"
+	"github.com/prebid/prebid-server/v3/config"
+	"github.com/prebid/prebid-server/v3/openrtb_ext"
+	"github.com/prebid/prebid-server/v3/util/jsonutil"
+	"github.com/prebid/prebid-server/v3/util/ptrutil"
+	"github.com/prebid/openrtb/v20/openrtb2"
+	"github.com/prebid/prebid-server/v3/adapters"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestJsonSamples(t *testing.T) {
+	bidder, buildErr := Builder(
+		openrtb_ext.BidderRiseMediaTech,
+		config.Adapter{
+			Endpoint: "https://dev-ads.risemediatech.com/ads/rtb/prebid/server",
+		},
+		config.Server{
+			ExternalUrl: "http://hosturl.com",
+			GvlID:       0,
+			DataCenter:  "2",
+		},
+	)
+
+	if buildErr != nil {
+		t.Fatalf("Builder returned unexpected error: %v", buildErr)
+	}
+	require.NoError(t, buildErr, "Builder returned unexpected error")
+
+	adapterstest.RunJSONBidderTest(t, "risemediatechtest", bidder)
+}
+
+// Table-driven test for parseImpExt
+func TestParseImpExt(t *testing.T) {
+	tests := []struct {
+		name    string
+		ext     jsonutil.RawMessage
+		wantErr bool
+	}{
+		{"Valid ext", jsonutil.RawMessage(`{"bidder":{"placementId":"abc"}}`), false},
+		{"Invalid JSON", jsonutil.RawMessage(`not-json`), true},
+		{"Not an object", jsonutil.RawMessage(`"string"`), true},
+		{"Bidder not object", jsonutil.RawMessage(`{"bidder":"not-an-object"}`), true},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			_, err := parseImpExt(tt.ext)
+			if tt.wantErr && err == nil {
+				assert.Error(t, err)
+			}
+			if !tt.wantErr && err != nil {
+				assert.NoError(t, err)
+			}
```

Comment:
The logic seems left over from before using assertions… the require will exit early and the assertion, itself, is checking for nil-ness so you don't have to. The only nuance is that if you want an error, you most likely need to return after you get one (because require will pass). Then you can treat it like a pragmatic guard where you just continue with remaining "else" case without an "else" clause
```suggestion
			if tt.wantErr {
				require.Error(t, err)
				return
			}
			require.NoError(t, err)
```